nerfstudio/model_components/renderers.py

```diff
@@ -101,26 +101,64 @@ def combine_rgb(
             comp_rgb = torch.sum(weights * rgb, dim=-2)
             accumulated_weight = torch.sum(weights, dim=-2)
 
+        background_color = cls.get_background_color(rgb=rgb, background_color=background_color)
+        comp_rgb = comp_rgb + background_color.to(weights.device) * (1.0 - accumulated_weight)
+
+        return comp_rgb
+
+    @classmethod
+    def get_background_color(cls, rgb: Float[Tensor, "*bs num_samples 3"], background_color: BackgroundColor) -> Tensor:
+        """Returns the RGB background color for a specified background color.
+
+        Args:
+            rgb: RGB for each sample.
+            background_color: The background color specification.
+
+        Returns:
+            Background color as RGB.
+        """
         if BACKGROUND_COLOR_OVERRIDE is not None:
             background_color = BACKGROUND_COLOR_OVERRIDE
-        if background_color == "last_sample":
+        elif background_color == "last_sample":
             background_color = rgb[..., -1, :]
-        if background_color == "random":
-            background_color = torch.rand_like(comp_rgb).to(rgb.device)
-        if isinstance(background_color, str) and background_color in colors.COLORS_DICT:
+        elif background_color == "random":
+            background_color = torch.rand_like(rgb[..., -1, :]).to(rgb.device)
+        elif isinstance(background_color, str) and background_color in colors.COLORS_DICT:
             background_color = colors.COLORS_DICT[background_color].to(rgb.device)
+        assert isinstance(background_color, Tensor)
+        return background_color
 
-        assert isinstance(background_color, torch.Tensor)
-        comp_rgb = comp_rgb + background_color.to(weights.device) * (1.0 - accumulated_weight)
+    @classmethod
+    def blend_background(
+        cls,
+        image: Tensor,
+        outputs: Dict[str, Union[Tensor, list]],
```

Comment:
Can you please pass the optional background_color directly instead of passing the outputs?

Reply:
Also, can you please pass RGB image and "opacity" as two separate arguments?

Reply:
Passed them as separate arguments. Also, it is probably better to not allow for optional arguments but only call this function if we actually want to blend the background. We can resolve this if you agree with removing the optionality of the arguments.

Reply:
But shouldn't blending happen always? When background_color=random it would use the provided background_color, otherwise it would use the renderer's background colour? Perhaps I am missing something...

Reply:
It should only happen if the input data includes an alpha channel. If this is the case, we would also have a `background_color` to pass to this method.
So you are right, `if "background_color" in outputs` is not needed.

Reply:
Actually, I just wanted to change this and realized that using the default `background_color` is non-trivial without saving extra data somewhere. For example, at the moment, we cannot get the RGB color of "last_sample" at this point in the code.

Do you see a specific use case for this functionality (alpha channel for blending without alpha transparent training)? If not, I'd prefer to leave it to keep the code cleaner.

Reply:
Ok, let's keep it cleaner. The reason why I suggested this is that if we always load the alpha channel and background_color!=random, there would still be the alpha channel in the target image, no?

Reply:
Yes, but in this case, we would want to discard the alpha channel as we don't currently use it for anything other than alpha-transparent training. At least I don't see the benefit of always blending the background color if an alpha channel is present.

Reply:
I remember looking at this a good while ago and thinking that a scoreboard of transient values, like this run's background color selection, could be more generally useful.

The other use case I had in my head was some way to prevent single bad iterations from having a catastrophic impact, maybe pause/stop training if metrics plummet instead of saving checkpoint.

Reply:
What are you thinking in terms of scoreboard? Do you mean having a dedicated object/place to write values like `background_color` to instead of storing them in `outputs`? 

>The other use case I had in my head was some way to prevent single bad iterations from having a catastrophic impact, maybe pause/stop training if metrics plummet instead of saving checkpoint.

I'm sorry, but I'm not quite following how this relates to always blending the background color if an alpha channel is present. Or do you mean as a use case for the scoreboard idea?

Reply:
> Do you mean having a dedicated object/place to write values like background_color to instead of storing them in outputs?

That was the thought, although not sure whether there is need.

> Or do you mean as a use case for the scoreboard idea?

Yes, this (sorry)

Reply:
This seems like something that would warrant a separate pull request that then also implements one of your suggested functionalities.
However, I do see one reason for implementing something like this as part of this PR: All the outputs are currently passed through [this resizing operation](https://github.com/nerfstudio-project/nerfstudio/blob/main/nerfstudio/models/base_model.py#L187). This doesn't make much sense for `background_color`. It does not crash as the background color is a per-image pixel color, but it still isn't great.

Any opinions on this or separate PR?

Reply:
Separate PR probably - unless it would fit neatly into this one